netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +203,11 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new Inbox.BoutNotFoundException(this.bout);
```

Comment:
@original-brownbear i agree about wrapping of whole set of directives, but i don't like `HttpException` in `DyMessages`(i don't like and `Inbox.BoutNotFoundException` actually, i would prefer separated `BoutNofFoundException`). `DyMessages` doesn't know about our Http environment, we mix our http business to database layer.